osh/cmd_eval.py

```diff
@@ -1030,6 +1030,11 @@ def _DoMutation(self, node):
                             key = val_ops.ToStr(lval.index,
                                                 'Obj index should be Str',
                                                 loc.Missing)
+                            # Hack: let SearchPath know it needs to reset the path
+                            # cache before looking up a command
+                            if key == 'PATH':
```

Comment:
Hm let me think about this  ... it seems a bit overzealous,  since it also clears the cache on `setvar other.PATH = 42`, I think

We could define YSH so you do `hash -r` aka `hash --clear` or something ... although maybe that's too unfriendly